Mathlib/Topology/MetricSpace/GromovHausdorffRealized.lean

```diff
@@ -0,0 +1,548 @@
+      _ = maxVar X Y * 1 := by simp
+      _ ≤ maxVar X Y * dist (inl x) (inr y) :=
+        mul_le_mul_of_nonneg_left Sum.one_le_dist_inl_inr (le_trans zero_le_one (one_le_maxVar X Y))
+  | inr x, inr y =>
+    calc
+      f (inr x, inr y) = dist x y := candidates_dist_inr fA x y
+      _ = dist (inr x) (inr y) := by
+        rw [@Sum.dist_eq X Y]
+        rfl
+      _ = 1 * dist (inr x) (inr y) := by simp
+      _ ≤ maxVar X Y * dist (inr x) (inr y) :=
+        mul_le_mul_of_nonneg_right (one_le_maxVar X Y) dist_nonneg
+
+/-- Technical lemma to prove that candidates are Lipschitz -/
+private theorem candidates_lipschitz_aux (fA : f ∈ candidates X Y) :
+    f (x, y) - f (z, t) ≤ 2 * maxVar X Y * dist (x, y) (z, t) :=
+  calc
+    f (x, y) - f (z, t) ≤ f (x, t) + f (t, y) - f (z, t) :=
+      sub_le_sub_right (candidates_triangle fA) _
+    _ ≤ f (x, z) + f (z, t) + f (t, y) - f (z, t) :=
+      (sub_le_sub_right (add_le_add_right (candidates_triangle fA) _) _)
+    _ = f (x, z) + f (t, y) := by simp [sub_eq_add_neg, add_assoc]
+    _ ≤ maxVar X Y * dist x z + maxVar X Y * dist t y :=
+      (add_le_add (candidates_dist_bound fA) (candidates_dist_bound fA))
+    _ ≤ maxVar X Y * max (dist x z) (dist t y) + maxVar X Y * max (dist x z) (dist t y) := by
+      apply add_le_add
+      apply mul_le_mul_of_nonneg_left (le_max_left (dist x z) (dist t y))
+        (zero_le_one.trans (one_le_maxVar X Y))
+      apply mul_le_mul_of_nonneg_left (le_max_right (dist x z) (dist t y))
+        (zero_le_one.trans (one_le_maxVar X Y))
+    _ = 2 * maxVar X Y * max (dist x z) (dist y t) := by
+      rw [dist_comm t y]
+      ring
+    _ = 2 * maxVar X Y * dist (x, y) (z, t) := by rfl
+
+/-- Candidates are Lipschitz -/
+private theorem candidates_lipschitz (fA : f ∈ candidates X Y) :
+    LipschitzWith (2 * maxVar X Y) f := by
+  apply LipschitzWith.of_dist_le_mul
+  rintro ⟨x, y⟩ ⟨z, t⟩
+  rw [Real.dist_eq, abs_sub_le_iff]
+  use candidates_lipschitz_aux fA
+  rw [dist_comm]
+  exact candidates_lipschitz_aux fA
+
+/-- candidates give rise to elements of `BoundedContinuousFunction`s -/
+def candidatesBOfCandidates (f : ProdSpaceFun X Y) (fA : f ∈ candidates X Y) : Cb X Y :=
+  BoundedContinuousFunction.mkOfCompact ⟨f, (candidates_lipschitz fA).continuous⟩
+#align Gromov_Hausdorff.candidates_b_of_candidates GromovHausdorff.candidatesBOfCandidates
+
+theorem candidatesBOfCandidates_mem (f : ProdSpaceFun X Y) (fA : f ∈ candidates X Y) :
+    candidatesBOfCandidates f fA ∈ candidatesB X Y :=
+  fA
+#align Gromov_Hausdorff.candidates_b_of_candidates_mem GromovHausdorff.candidatesBOfCandidates_mem
+
+/-- The distance on `X ⊕ Y` is a candidate -/
+private theorem dist_mem_candidates :
+    (fun p : (X ⊕ Y) × (X ⊕ Y) => dist p.1 p.2) ∈ candidates X Y := by
+  simp only [candidates, dist_comm, forall_const, and_true_iff, add_comm, eq_self_iff_true,
+    and_self_iff, Sum.forall, Set.mem_setOf_eq, dist_self]
+  repeat'
```

Comment:
I removed the `repeat'` and manually applying the tactics. This is the required order 
```lean
  constructor
  constructor
  constructor
  exact fun x y => by rfl
  exact fun x y => by rfl
  constructor
  exact fun x y => maxVar_bound
```

The error appears in the last step: `  exact fun x y => maxVar_bound`

```lean
typeclass instance problem is stuck, it is often due to metavariables
 Nonempty (?m.99960 x y)
```

I don't know enough about this file to fix this.

Reply:
Actually, the last line is supposed to be `  exact fun a y z => dist_triangle_left _ _ _` which throws the error

```lean 
typeclass instance problem is stuck, it is often due to metavariables
 PseudoMetricSpace (?m.99752 a y z)
```

:shrug:

Reply:
I found a new proof that uses less complicated things